components/locid/src/subtags/region.rs

```diff
@@ -61,6 +62,37 @@ impl Region {
         }
     }
 
+    /// Safely creates a [`Region`] from a reference to its raw format
+    /// as returned by [`Region::into_raw()`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::subtags::Region;
+    ///
+    /// assert!(matches!(Region::try_from_raw(b"US\0"), Ok(_)));
+    /// assert!(matches!(Region::try_from_raw(b"419"), Ok(_)));
+    /// assert!(matches!(Region::try_from_raw(b"foo"), Err(_)));
+    ///
+    /// // Unlike the other constructors, this one is case-sensitive:
+    /// assert!(matches!(Region::try_from_raw(b"us\0"), Err(_)));
+    /// ```
+    pub fn try_from_raw(v: &[u8; 3]) -> Result<&Self, ParserError> {
+        let s = TinyAsciiStr::<{ core::mem::size_of::<Self>() }>::try_from_raw(v)
+            .map_err(|_| ParserError::InvalidSubtag)?;
+        let is_valid = match s.len() {
+            REGION_ALPHA_LENGTH => s.is_ascii_uppercase(),
+            REGION_NUM_LENGTH => s.is_ascii_numeric(),
+            _ => false,
+        };
+        if is_valid {
+            // Safe since the bytes are valid
```

Comment:
nit:
```suggestion
            // Safe since the bytes are valid and Self is transparent over TinyAsciiStr<3>
```

Reply:
This is obsolete since now that I return by value I can use a constructor rather than a transmute.